File: api/booking-service/src/main/java/com/carrental/bookingservice/model/dto/BookingAddRequestDTO.java
package com.carrental.bookingservice.model.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import javax.validation.constraints.NotNull;
import java.time.LocalDate;

@Data
public class BookingAddRequestDTO {

    @NotNull
    private Long vehicleId;

    @NotNull
    private Long locationId;

    @NotNull
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    private LocalDate receiptDate;

    @NotNull
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    private LocalDate returnDate;
}

File: frontend-app/reservation-page/src/model/BookingAddRequestDTO.ts
class BookingAddRequestDTO {
    locationId: number;
    vehicleId: number;
    receiptDate: string;
    returnDate: string;

    constructor(locationId: number, vehicleId: number, receiptDate: string, returnDate: string) {
        this.locationId = locationId;
        this.vehicleId = vehicleId;
        this.receiptDate = receiptDate;
        this.returnDate = returnDate;
    }
}

export default BookingAddRequestDTO;
